resources/cmake/ConnextDdsAddSecurityArtifacts.cmake

```diff
@@ -0,0 +1,186 @@
+# (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+# No duplications, whole or partial, manual or electronic, may be made
+# without express written permission.  Any such copies, or revisions thereof,
+# must display this notice unaltered.
+# This code contains trade secrets of Real-Time Innovations, Inc.
+
+#[[.rst:
+.. _connextdds_add_security_artifacts:
+
+ConnextDdsAddSecurityArtifacts
+------------------------------
+
+Generate the artifacts needed for secure examples.
+
+``connextdds_add_security_artifacts()``
+
+It is based on:
+ - The OpenSSL configuration files in the resources/security/<ca> directory.
+
+ - The Governance and Permissions files in the resources/xml directory.
+
+So far this means:
+
+ - One ECDSA CA and two peers.
+
+ - RTPS and metadata protection kind set to ENCRYPT.
+
+ - Publishing and subscribing to any domain and topic is allowed, only for those
+   peers.
+
+#]]
+
+include(UseOpenSSL)
+
+
+function(connextdds_add_security_artifacts)
+    # Directory where the security artifacts are.
+    set(SECURITY_RESOURCES_PATH
+        ${SECURITY_RESOURCES_PATH}
+        "${CMAKE_MODULE_PATH}/../security"
+    )
+
+    # ##########################################################################
+    # Generate the structure for the directory with the security artifacts.
+    # <example/<CMAKE_CURRENT_BINARY_DIR>/security/ecdsa01
+    #   |- certs/
+    #   |- temporary_files/
+    #   |- xml/
+    #   |   |- signed/
+    #
+    # ##########################################################################
+    # security.
+    set(security_root_directory
+        ${security_root_directory}
+        "${CMAKE_CURRENT_BINARY_DIR}/security"
+    )
+    file(MAKE_DIRECTORY ${security_root_directory})
```

Comment:
Avoid creating files during the build system generation. It is a better idea to create a [CMake custom command](https://cmake.org/cmake/help/v3.11/command/add_custom_command.html#add-custom-command) to create all the files and make the target `securityArtifacts `dependent` of the output of the CMake custom command.

[This post can help you](https://dev.to/iblancasa/learning-cmake-3-understanding-addcustomcommand-and-addcustomtarget-43gp).

Reply:
I added the custom command. I had to modify the code a bit to get it to work. We can talk about it later in the meeting.

Thanks @iblancasa.